pkg/inject/envoy.go

```diff
@@ -107,7 +108,13 @@ func (m *envoyMutator) mutate(pod *corev1.Pod) error {
 	if m.mutatorConfig.enableSDS && !isSDSDisabled(pod) {
 		mutateSDSMounts(pod, &container, m.mutatorConfig.sdsUdsPath)
 	}
-	pod.Spec.Containers = append(pod.Spec.Containers, container)
+
+	if m.mutatorConfig.waitUntilProxyStarts {
+		pod.Spec.Containers = append([]corev1.Container{container}, pod.Spec.Containers...)
```

Comment:
Is there any reason we should not always add Envoy to the front of the list? Can we get rid of this conditional and reduce testing scope?

Reply:
the only thing i can think of is that some kubectl commands like kubectl exec default to the first app container and customers may find it strange to see the order changed due to the injection. does anyone else know other implications for changing the order @ar4096 @bendu @Y0Username @joesbigidea ? that being said, it may be safer to tie this behavior to the feature flag in the situation that there are other implications

Reply:
We should think on this some more @paulyeo21. Remember networking is already broken because the init container already redirected traffic to our sidecar, so whether _waitUntilReady_ is enabled or not, we still might want to be the first container out of the gate.

Reply:
i understand that the networking is currently "broken" and we want to make this feature's behavior as the default, but i would still prefer to leave the default to be existing behavior and have the changes introduced in this PR to be toggle-able since this introduces a lot of change